src/services/CertificateGenerationService.ts

```diff
@@ -191,9 +193,26 @@ class CertificateGenerationService {
       Signature: {
         ImageType: "png",
         ImageData: signature,
-      },
+      }
     };
-    const { testTypes, vehicleType, systemNumber } = testResult;
+
+    const { testTypes, vehicleType, systemNumber, testHistory } = testResult;
+
+    if (testHistory) {
+      for (const history of testHistory) {
+        for (const testType of history.testTypes) {
+          if (testType.testCode === testTypes.testCode) {
+            payload.Reissue = {
```

Comment:
Is it possible that we have a requirement for creating replacements on linked tests where more than one test is held within a test result record. 
This code is saying we can only have one reissue per test result record.

Reply:
I think I understand your query. The cert-gen-init functions sends all the tests in a test record as separate messages. Cert-gen will only ever see a single test at a time.

Reply:
Ahh amazing